src/contexts/AuthUserContext/index.tsx

```diff
@@ -25,6 +26,17 @@ export function AuthUserProvider({
   useEffect(() => {
     firebase.getAuth().onAuthStateChanged((authUser: AuthUser | null) => {
       authUser ? setAuthUser(authUser) : setAuthUser(null);
+      if (authUser) {
+        axios.interceptors.request.use(
+          async (request) => {
+            request.headers.authorization = await authUser.getIdToken(true);
+            return request;
+          },
+          (error) => {
+            return Promise.reject(error);
+          }
+        );
+      }
```

Comment:
I figured out a way for us to move the interceptor out of UserService. This `onAuthStateChange` gets called whenever the user logins, so then we will update the interceptor with the new idToken. Hypothetically, if the idToken expires, we will update the interceptor with the new idToken if we re-authenticate the user. 

I think we can test out token expiration in another ticket, but this should do for now.

Reply:
Hmm but onAuthStateChanged() will only be called when the user logs in and logs out, so the idToken would stay in the header even after expiry. I think it's better to call getIdToken() every time a request is made to the backend to ensure the token isn't expired.

Reply:
Oh I see, kk thanks that makes sense. This post is helpful: https://stackoverflow.com/questions/49656489/is-the-firebase-access-token-refreshed-automatically/49697718#49697718. 

https://github.com/ubclaunchpad/PromoPal-frontend/pull/41/commits/1d32928516651ede08abcbedfd67af4c1ec14237
I've made it now so that each request to BE that requires an idToken will call firebase getIdToken(). If the token has expired, getIdToken will refresh the token and return a new one. This will trigger `onAuthStateChange` which will then update the interceptor with the newest idToken. That way as developers, we don't need to be responsible for always adding the authorization header, we just need to know to call the token each time we make a request.

Reply:
https://github.com/ubclaunchpad/PromoPal-frontend/pull/41/commits/de81cc1976e810896fdab72baac711be643b4d02
Also changed here, `authUser.getIdToken(true);` we don't need to specify true.